### PR TITLE
Item Number Mismatch

### LINKED ISSUE #2564: Adjust TANF and SSP Section 1 item numbers per OFA guidance
**Description:**
Alex found via [this comment](https://github.com/raft-tech/TANF-app/pull/2504/files/a31b32f9d2b6b0a582bca87db6487b5adaacbc33#r1218263890) that we are using the wrong item numbers. It appears we did not consult the correct ordering. Please see document section below for updated excel spreadsheets which directly link item number and description and update our RowSchema accordingly.

As a data analyst, I need TANF and SSP Section 1 item numbers to match OFA documentation.

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [ ] All displayed error messages should output items numbers that map to the supporting documentation.

- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**

- [x] Will need excel docs for everything not T1 - @ADPennington 
- [ ] Adjust item numbers in RowSchema for all SSP Section 1
- [ ] Adjust item numbers in RowSchema for all TANF Section 1
- [ ] Spike: analyze any locations that these updated item numbers could conflict (like in t1 models in prod)
- [ ] Coordinate with Alex to ensure prod item numbers are overwritten in an automated fashion.
- [ ] Adjust any tests as needed
- [ ] STRETCH: Look for modular way to update these item numbers in the future.
- [ ] Run Testing Checklist and confirm all tests pass

**Notes:**
- Please avoid migration(s) if possible.

**Supporting Documentation:**

- [T1 sheet](https://gorafttech.sharepoint.com/:x:/r/sites/TDRSResearchDesign/_layouts/15/Doc.aspx?sourcedoc=%7BF99CEA75-18CC-4494-9302-677CC92E3EF4%7D&file=TDP_Validation.xlsx&action=default&mobileredirect=true)
- [SSP, M1-M3](https://gorafttech.sharepoint.com/:x:/r/sites/TDRSResearchDesign/_layouts/15/Doc.aspx?sourcedoc=%7BD750ECCD-B320-4E7A-B7F6-C9354B84F61B%7D&file=TDP_Validation_SSP_Section1.xlsx&action=default&mobileredirect=true)
- [99 page doc](https://gorafttech.sharepoint.com/:w:/r/sites/TDRSResearchDesign/_layouts/15/Doc.aspx?sourcedoc=%7BFD51E66D-6DDA-47AA-A970-29006EEC07F3%7D&file=TANF-SSP%20data%20instructions_clean.docx&action=default&mobileredirect=true)

**Open Questions:**

- Update these values in stored ParserError objects in prod DB?

### PR DESCRIPTION
**Pull request by elipe17**
## Summary of Changes
- Updated RowSchema and Models to reference correct item number
- Converted item number to string to capture subcategories in certain items
- Added missing field to M1 schema
Pull request closes #2564 

## How to Test
```
cd tdrs-frontend && docker-compose up
cd tdrs-backend && docker-compose up
```
- Authenticate as a user who can submit both tanf and ssp datafiles
- Submit a valid and invalid file. Verify the valid files records were created and that the appropriate ParserError objects were created for the invalid file(s). In the ParserError object, verify that the item number is correct wrt [the references here](https://www.acf.hhs.gov/ofa/policy-guidance/final-tanf-ssp-moe-data-reporting-system-transmission-files-layouts-and-edits)

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] Item numbers are correctly referenced

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [x] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

**Review by jtimpe — APPROVED** [review]

**Review by raftmsohani — APPROVED** [review]
LGTM! Just be careful with migration, maybe run it on a clone of production to see the impact first

File: tdrs-backend/tdpservice/parsers/schema_defs/ssp/m2.py [hunk 1]
```diff
@@ -13,137 +13,137 @@
...
+        Field(item="45", name='SUB_PUBLIC_EMPLOYMENT', type='number', startIndex=68, endIndex=70,
               required=True, validators=[]),
-        Field(item=33, name='WORK_EXPERIENCE_HOP', type='number', startIndex=70, endIndex=72,
+        Field(item="46A", name='WORK_EXPERIENCE_HOP', type='number', startIndex=70, endIndex=72,
               required=True, validators=[]),
-        Field(item=34, name='WORK_EXPERIENCE_EA', type='number', startIndex=72, endIndex=74,
+        Field(item="46B", name='WORK_EXPERIENCE_EA', type='number', startIndex=72, endIndex=74,
               required=True, validators=[]),
-        Field(item=35, name='WORK_EXPERIENCE_HOL', type='number', startIndex=74, endIndex=76,
+        Field(item="47C", name='WORK_EXPERIENCE_HOL', type='number', startIndex=74, endIndex=76,
```
Comment (ADPennington) [inline tdrs-backend/tdpservice/parsers/schema_defs/ssp/m2.py]:
```suggestion
        Field(item="46C", name='WORK_EXPERIENCE_HOL', type='number', startIndex=74, endIndex=76,
```

File: tdrs-backend/tdpservice/parsers/schema_defs/ssp/m2.py [hunk 2]
```diff
@@ -13,137 +13,137 @@
...
               required=True, validators=[]),
-        Field(item=36, name='OJT', type='number', startIndex=76, endIndex=78,
+        Field(item="48", name='OJT', type='number', startIndex=76, endIndex=78,
```
Comment (ADPennington) [inline tdrs-backend/tdpservice/parsers/schema_defs/ssp/m2.py]: @elipe17 this is 47, and so on thereafter this. [reference](https://gorafttech.sharepoint.com/:x:/r/sites/TDRSResearchDesign/_layouts/15/Doc.aspx?sourcedoc=%7BD750ECCD-B320-4E7A-B7F6-C9354B84F61B%7D&file=TDP_Validation_SSP_Section1.xlsx&action=default&mobileredirect=true)

File: tdrs-backend/tdpservice/parsers/schema_defs/tanf/t1.py [hunk 3]
```diff
@@ -13,96 +13,96 @@
...
+        Field(item="26AV", name='NON_COOPERATION_CSE', type='number', startIndex=99, endIndex=100,
               required=True, validators=[]),
-        Field(item=36, name='FAILURE_TO_COMPLY', type='number', startIndex=100, endIndex=101,
+        Field(item="26AVI", name='FAILURE_TO_COMPLY', type='number', startIndex=100, endIndex=101,
               required=True, validators=[]),
-        Field(item=37, name='OTHER_SANCTION', type='number', startIndex=101, endIndex=102,
+        Field(item="26AVII", name='OTHER_SANCTION', type='number', startIndex=101, endIndex=102,
               required=True, validators=[]),
-        Field(item=38, name='RECOUPMENT_PRIOR_OVRPMT', type='number', startIndex=102, endIndex=106,
+        Field(item="26BI", name='RECOUPMENT_PRIOR_OVRPMT', type='number', startIndex=102, endIndex=106,
```
Comment (ADPennington) [inline tdrs-backend/tdpservice/parsers/schema_defs/tanf/t1.py]:
```suggestion
        Field(item="26B", name='RECOUPMENT_PRIOR_OVRPMT', type='number', startIndex=102, endIndex=106,
```

File: tdrs-backend/tdpservice/parsers/schema_defs/tanf/t1.py [hunk 4]
```diff
@@ -13,96 +13,96 @@
...
               required=True, validators=[]),
-        Field(item=39, name='OTHER_TOTAL_REDUCTIONS', type='number', startIndex=106, endIndex=110,
+        Field(item="26BII", name='OTHER_TOTAL_REDUCTIONS', type='number', startIndex=106, endIndex=110,
```
Comment (ADPennington) [inline tdrs-backend/tdpservice/parsers/schema_defs/tanf/t1.py]: @elipe17 this should be 26ci. everything after this needs an update. see [here](https://gorafttech.sharepoint.com/:x:/r/sites/TDRSResearchDesign/_layouts/15/Doc.aspx?sourcedoc=%7BF99CEA75-18CC-4494-9302-677CC92E3EF4%7D&file=TDP_Validation_TANF_Section1.xlsx&action=default&mobileredirect=true)

**Review by ADPennington — CHANGES_REQUESTED** [review]
@elipe17 looks good; just a couple of the schemas need updated item numbers. 

Also can you confirm, should I see item numbers in the schemas in DAC?

**Review by andrew-jameson — COMMENTED** [review]
Besides the ordering issues Alex brought up, this looks straight-forward. Will test once those updates are live.

**Comment by elipe17** [comment]
> @elipe17 looks good; just a couple of the schemas need updated item numbers.
> 
> Also can you confirm, should I see item numbers in the schemas in DAC?

Updated the item numbers. And I apologize, you will not see the item numbers in DAC.

**Review by ADPennington — APPROVED** [review]
🚀

**Pull request merged**